File: 1468-check-if-n-and-its-double-exist/check-if-n-and-its-double-exist.cpp
class Solution {
public:
    bool checkIfExist(vector<int>& arr) {
        map<int,int> mp;
        for(auto it: arr) mp[it]++;
        for(auto it: arr){
            if(it==2*it && mp[it]>1) return true;
            if(it!=2*it && mp[2*it]) return true;
        }
        return false;
    }
};

File: 1468-check-if-n-and-its-double-exist/check-if-n-and-its-double-exist.java
class Solution {
    public boolean checkIfExist(int[] arr) {
        Map<Integer, Integer> mp = new HashMap<>();
        for(int it: arr){
            mp.put(it, mp.getOrDefault(it, 0)+1);
        }
        for(int it: arr){
            if(it==2*it && mp.getOrDefault(it,0)>1) return true;
            if(it!=2*it && mp.getOrDefault(2*it,0)>=1) return true;

        }
        return false;
    }
}